extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/TradfriColor.java

```diff
@@ -279,5 +279,21 @@ public static PercentType calculateColorTemperature(int xyX, int xyY) {
         }
         return new PercentType((int) Math.round(value * 100.0));
     }
+    
+    
+    /**
+     * Converts the xyBrightness value to PercentType
+     *
+     * @param xyBrightness xy brightness level 0 to 254
+     * @return {@link PercentType} with brightness level (0 = light is off, 1 = lowest, 100 = highest)
+     */
+    public static PercentType xyBrightnessToPercentType(int xyBrightness) {
+        if (xyBrightness > 254) {
```

Comment:
out of curiosity: why are values > 254 a problem? All we have to do is adopt the `2.54` in line 296 and we could also deal with 255 or greater. wdyt?

Reply:
Well - my understanding for a PercentType is to be between 0 and 100. So - if `xyBrightness > 254` then `xyBrightness / 2.54` will return value `> 100` and it will throw `IllegalArgumentException`

Reply:
Yes I get that. All we have to adopt is the divisor (2.54) to a value that matches our upper bound (254). In case we move the upper bound to 255 the divisor should be adopted to 2.55 to match the PercentType. I just wanted to know where the 254 originates and why there are possible values grater then 254.

Reply:
I guess this comes from the IKEA API (not not really sure)

May be @sjka or @kaikreuzer can give more info

Reply:
Yes, the range 0 to 254 is defined for Zigbee ZLL brightness and the IKEA gateway adopts this.